### PR TITLE
More Stdlib + Misc

### PR DESCRIPTION
**Pull request by HoshinoTented**
* `Reflect` (Coq style)
* `=? : Nat -> Nat -> Bool` and its properties
* more lemmas for `Bool`
* an error for `overlap def foo => bar`

The last change looks lonely if I create a new PR for it, so I put them together.

### CODE REVIEW

**Comment by ice1000** [comment]
It's better to define =? as decEq which returns decidable. In fact I think it is in the library

File: base/src/test/resources/success/common/src/Arith/Nat/Properties.aya [hunk 1]
```diff
@@ -10,3 +16,42 @@ def z≠s {a : Nat} (p : 0 = suc a) : False => (\i => diag (p i)).coe a
...
+
+def =?-= {m n : Nat} (p : (m =? n) = true) : m = n
+| {0}, {0}, _ => idp
+| {0}, {suc n'}, p => elim (t≠f (sym p))
+| {suc m'}, {0}, p => elim (t≠f (sym p))
+| {suc m'}, {suc n'}, p =>
+  // p : (suc m' =? suc n') = true
+  //   : (m' =? n') = true            simpl.
+  // Goal : suc m' = suc n'
+  pmap suc (=?-= {m'} {n'} p)
```
Comment (ice1000) [inline base/src/test/resources/success/common/src/Arith/Nat/Properties.aya:40]: Is it possible to do it w/o these implicits?

File: base/src/test/resources/success/common/src/Arith/Nat/Core.aya [hunk 2]
```diff
@@ -60,3 +63,10 @@ private overlap def *-assoc-lemma : Associative (*)
 | 0, _, _ => idp
 | suc m, _, _ => <=3=> (+-*-distl _) (pmap (+ _) *-assoc) +-comm
 inline def *-assoc {m n k : Nat} => *-assoc-lemma m n k
+
+overlap def infix =? (m n : Nat) : Bool
```
Comment (ice1000) [inline base/src/test/resources/success/common/src/Arith/Nat/Core.aya:67]: This definition is not overlapping. I'll try to write a warning that detects such non-overlapping `overlap` definitions.

**Comment by HoshinoTented** [comment]
> It's better to define =? as decEq which returns decidable. In fact I think it is in the library

What is the math symbol of `decEq`? I am not friendly with the Agda-Style library.

File: base/src/test/resources/success/common/src/Arith/Nat/Properties.aya (hunk 1 above)
Comment (HoshinoTented) [reply base/src/test/resources/success/common/src/Arith/Nat/Properties.aya:40]: 🤔what is `w/o`?

**Comment by ice1000** [comment]
> > It's better to define =? as decEq which returns decidable. In fact I think it is in the library
> 
> What is the math symbol for `decEq`? I am not friendly with the Agda-Style library.

You don't have to always use unicode, we can use English names

File: base/src/test/resources/success/common/src/Arith/Nat/Equality.aya [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+open import Arith::Nat::Properties
+open import Logic::Dec
+open import Logic::False
+
+def ≐-suc {m n : Nat} (p : Dec (m = n)) : Dec (suc m = suc n)
+| yes p => yes (pmap suc p)
+| no np => no (\ p =>
+  // p  : suc m = suc n
+  // np : m = n -> False
+  // Goal : False
```
Comment (ice1000) [inline base/src/test/resources/success/common/src/Arith/Nat/Equality.aya:12]: Are these information available in vscode?

File: base/src/test/resources/success/common/src/Arith/Nat/Properties.aya (hunk 1 above)
Comment (ice1000) [reply base/src/test/resources/success/common/src/Arith/Nat/Properties.aya]: It's 'without'

File: base/src/test/resources/success/common/src/Arith/Nat/Equality.aya (hunk 3 above)
Comment (HoshinoTented) [reply base/src/test/resources/success/common/src/Arith/Nat/Equality.aya:12]: No, I list them manually for me.

Comment (ice1000) [reply base/src/test/resources/success/common/src/Arith/Nat/Equality.aya:12]: I think they are

Comment (ice1000) [reply base/src/test/resources/success/common/src/Arith/Nat/Equality.aya:12]: Use a hole, `{??}` as an expression

File: base/src/test/resources/success/common/src/Arith/Nat/Properties.aya (hunk 1 above)
Comment (HoshinoTented) [reply base/src/test/resources/success/common/src/Arith/Nat/Properties.aya]: These comments can be deleted. But it helps someone who reads this code.

**Comment by ice1000** [comment]
Good
bors r+

**Pull request merged**